mockgen/model/model.go

```diff
@@ -147,19 +149,58 @@ func init() {
 
 // ArrayType is an array or slice type.
 type ArrayType struct {
-	Len  int // -1 for slices, >= 0 for arrays
+	Len  Type // LengthType
```

Comment:
Although I doubt people use this directly, this is a breaking change. Does it need to be `Type` and not `int`?

Reply:
bump

Reply:
Not the author, but I think `int` wouldn't solve the purpose here. Because in case length is provided as a `const`, we need to store references to it; namely, `Package` and `Name` ([see here](https://github.com/jkmar/mock/blob/7f3c6493972e4d3593ba098b31d6abc58bd0f7c3/mockgen/parse.go#L457)).

I can take up this PR, if needed.

cc @codyoss